assets/js/blocks/checkout/inner-blocks/checkout-actions-block/block.tsx

```diff
@@ -7,21 +7,36 @@ import {
 	PlaceOrderButton,
 	ReturnToCartButton,
 } from '@woocommerce/base-components/cart-checkout';
+import { useCheckoutSubmit } from '@woocommerce/base-context/hooks';
+import { __experimentalApplyCheckoutFilter } from '@woocommerce/blocks-checkout';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
+import { defaultPlaceOrderButtonLabel } from './constants';
 
 const Block = ( {
 	cartPageId,
 	showReturnToCart,
 	className,
+	placeOrderButtonLabel,
 }: {
 	cartPageId: number;
 	showReturnToCart: boolean;
 	className?: string;
+	placeOrderButtonLabel: string;
 } ): JSX.Element => {
+	const { submitButtonText } = useCheckoutSubmit();
+	const defaultLabel =
+		submitButtonText ||
+		placeOrderButtonLabel ||
+		defaultPlaceOrderButtonLabel;
+	const label = __experimentalApplyCheckoutFilter( {
+		filterName: 'placeOrderButtonLabel',
+		defaultValue: defaultLabel,
```

Comment:
nitpick: you can probably inline the logic from line 31 above here to make it a bit clearer. We have a lot of things called `default...`: `defaultPlaceOrderButtonLabel`, `defaultLabel` and `defaultValue` all pointing to the same thing.

Reply:
I just changed 
```ts
const defaultLabel =
	paymentMethodButtonLabel ||
	placeOrderButtonLabel ||
	defaultPlaceOrderButtonLabel;
const label = __experimentalApplyCheckoutFilter( {
	filterName: 'placeOrderButtonLabel',
	defaultValue: defaultLabel,
} );
```
to
```ts
const label = __experimentalApplyCheckoutFilter( {
	filterName: 'placeOrderButtonLabel',
	defaultValue:
		paymentMethodButtonLabel ||
		placeOrderButtonLabel ||
		defaultPlaceOrderButtonLabel,
} );
```
and it's indeed more readable that way.